streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -993,12 +995,17 @@ private KafkaStreams(final TopologyMetadata topologyMetadata,
 
         // use client id instead of thread client id since this admin client may be shared among threads
         this.clientSupplier = clientSupplier;
-        adminClient = clientSupplier.getAdmin(applicationConfigs.getAdminConfigs(ClientUtils.adminClientId(clientId)));
+        final Map<String, Object> adminConfigs = applicationConfigs.getAdminConfigs(ClientUtils.adminClientId(clientId));
+        adminConfigs.put(AdminClientConfig.ENABLE_METRICS_PUSH_CONFIG, true);
```

Comment:
I think we should do this inside `getAdminConfigs(...)` and add `ADMIN_DEFAULT_OVERRIDES` similar to what we have for producer/consumer to `StreamsConfig`.

Plus add new test to `StreamsConfigTest` to cover the different edge cases, as describe on the KIP: https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=315494236#KIP1076:MetricsforclientapplicationsKIP714extension-Configuration

Reply:
Done via #17422